File: problems/1725-number-of-rectangles-that-can-form-the-largest-square/solution.cpp
class Solution {
public:
    int countGoodRectangles(vector<vector<int>>& rectangles) {
        int maxLen = 0;
        int count = 0;
        for (auto& rect : rectangles) {
            int side = min(rect[0], rect[1]);
            if (side > maxLen) {
                maxLen = side;
                count = 1;
            } else if (side == maxLen) {
                count++;
            }
        }
        return count;
    }
};

File: problems/1725-number-of-rectangles-that-can-form-the-largest-square/solution.py
class Solution:
    def countGoodRectangles(self, rectangles: List[List[int]]) -> int:
        maxLen = 0
        count = 0
        for l, w in rectangles:
            k = min(l, w)
            if k > maxLen:
                maxLen = k
                count = 1
            elif k == maxLen:
                count += 1
        return count